### PR TITLE
Trust and confusion minor bug fixes

### PR DESCRIPTION
**Pull request by naterush**
# Description

This PR fixes a large number of bugs that we found in the trust and confusion work. Specifically:

# Testing

Test these things:

    mitosheet: add a github link to signup flow
    mitosheet: hide scroll bars where they overlap with text
    mitosheet: log jupyterlab theme, number of sheets rendered
    mitosheet: make file browser work with arrow keys (and add loading message)
    mitosheet: make undo and redo work with sort, filter
    mitosheet: handle yes and no properly in string -> bool conversion
    mitosheet: make rename input work more naturally
    mitosheet: make graph params not clear if you select same data source
    mitosheet: disable filtering options on timedelta columns
    mitoinstaller: print out a message that it might take a few moments to install

# Documentation

Nope!

### CODE REVIEW

File: mitosheet/mitosheet/step_performers/sort.py [hunk 1]
```diff
@@ -61,7 +62,14 @@ def execute( # type: ignore
...
         try: 
             pandas_start_time = perf_counter()
-            new_df = prev_state.dfs[sheet_index].sort_values(by=column_header, ascending=(sort_direction == ASCENDING), na_position=('first' if sort_direction == ASCENDING else 'last'))
+            if sort_direction != SORT_DIRECTION_NONE:
+                new_df = prev_state.dfs[sheet_index].sort_values(by=column_header, ascending=(sort_direction == SORT_DIRECTION_ASCENDING), na_position=('first' if sort_direction == SORT_DIRECTION_ASCENDING else 'last'))
+            else:
+                # We notably let the user sort by the "none" direction, which effectively allows the user
+                # to unapply the sort by toggling the sort button after they click it once
+                new_df = prev_state.dfs[sheet_index]
+                print("NOT SORTING")
```
Comment (naterush) [inline mitosheet/mitosheet/step_performers/sort.py:71]: Here

File: mitosheet/src/components/elements/Dropdown.tsx [hunk 2]
```diff
@@ -367,7 +368,9 @@ const Dropdown = (props: DropdownProps): JSX.Element => {
                 className: classNames(child.props.className, {
                     'dropdown-item-selected': selected,
                     'dropdown-item-with-search-first-item': props.searchable === true && found === 0
-                })
+                }),
+                // TODO: this selected is wrong!
```
Comment (naterush) [inline mitosheet/src/components/elements/Dropdown.tsx:372]: Remove this!

File: mitosheet/src/components/endo/ColumnHeader.tsx [hunk 3]
```diff
@@ -350,6 +359,9 @@ const ColumnHeader = (props: {
                                     closeColumnHeaderEditor()
                                 }
                             }}
+                            onBlur={() => {
```
Comment (naterush) [inline mitosheet/src/components/endo/ColumnHeader.tsx:362]: Comment

File: mitosheet/src/components/icons/mitofolks/BlueMitoFolk.tsx [hunk 4]
```diff
@@ -1,124 +1,366 @@
...
-                    <stop stopColor="#0064AD"/>
-                    <stop offset="1" stopColor="white" stopOpacity="0"/>
-                </linearGradient>
-            </defs>
-        </svg>
-
-    )
-}
+const BlueMitoFolk = (): JSX.Element => (
+  <svg
```
Comment (naterush) [inline mitosheet/src/components/icons/mitofolks/BlueMitoFolk.tsx:4]: TODO: run lint

File: mitosheet/src/components/taskpanes/Import/FileBrowser.tsx [hunk 5]
```diff
@@ -1,97 +1,103 @@
...
-    }, [props.pathParts])
+        props.setImportState(prevImportState => {
+            return {
+                ...prevImportState,
+                searchString: ''
+            }
+        })
+    }, [props.importState.pathContents.path_parts])
+
+    // We make sure to always focus back on the search input
```
Comment (naterush) [inline mitosheet/src/components/taskpanes/Import/FileBrowser.tsx:51]: Update comment

File: mitosheet/src/components/taskpanes/Import/FileBrowser.tsx [hunk 6]
```diff
@@ -102,29 +108,66 @@ function FileBrowser(props: FileBrowserProps): JSX.Element {
                 </Col>
             </Row>
             <div className='mt-5px mb-5px'>
-                <Input
-                    value={search}
+                {/* NOTE: we use a raw input as we need to put a ref on it, which we don't support currently */}
```
Comment (naterush) [inline mitosheet/src/components/taskpanes/Import/FileBrowser.tsx:111]: TODO

File: mitosheet/src/components/taskpanes/Import/FileBrowser.tsx [hunk 7]
```diff
@@ -102,29 +108,66 @@ function FileBrowser(props: FileBrowserProps): JSX.Element {
...
+                        props.setImportState(prevImportState => {
+                            return {
+                                ...prevImportState,
+                                searchString: newSearchString
+                            }
+                        })
+                    }}
+                    onKeyDown={(e) => {
+                        if (e.key == 'ArrowUp') {
+                            // TODO: make it loop
```
Comment (naterush) [inline mitosheet/src/components/taskpanes/Import/FileBrowser.tsx:128]: Remove

File: mitosheet/src/components/taskpanes/Import/FileBrowserElement.tsx [hunk 8]
```diff
@@ -84,19 +86,30 @@ export const getInvalidFileError = (selectedElement: FileElement, excelImportEna
...
-    const isSelected = props.selectedElement !== undefined 
-        && props.element.isDirectory === props.selectedElement.isDirectory
-        && props.element.name === props.selectedElement.name;
+    const isSelected = props.index === props.importState.selectedElementIndex;
+
+    // If the element becomes selected, we make sure it is visible
+    useEffect(() => {
+        const element = elementRef.current;
+        const parent = elementRef.current?.parentElement;
+        console.log(parent)
```
Comment (naterush) [inline mitosheet/src/components/taskpanes/Import/FileBrowserElement.tsx:99]: Remove

File: mitosheet/src/components/taskpanes/Import/FileBrowserElement.tsx [hunk 9]
```diff
@@ -84,19 +86,30 @@ export const getInvalidFileError = (selectedElement: FileElement, excelImportEna
...
+        if (isSelected && element && parent) {
+            console.log("ENSURING in view")
```
Comment (naterush) [inline mitosheet/src/components/taskpanes/Import/FileBrowserElement.tsx:101]: And here

File: mitosheet/src/components/taskpanes/Import/ImportTaskpane.tsx [hunk 10]
```diff
@@ -159,11 +193,10 @@ function ImportTaskpane(props: ImportTaskpaneProps): JSX.Element {
             'selected_element_changed',
             {'selected_element': selectedElementName}
         )
-    }, [selectedElement])
-
 
-    const importButtonStatus = getImportButtonStatus(selectedElement, props.userProfile.excelImportEnabled);
+    }, [selectedElement])
 
+    const importButtonStatus = getImportButtonStatus(selectedElement, props.userProfile.excelImportEnabled, importState.loadingImport);
```
Comment (naterush) [inline mitosheet/src/components/taskpanes/Import/ImportTaskpane.tsx:199]: Cleanup this file!

File: mitosheet/src/components/taskpanes/Import/importUtils.tsx [hunk 11]
```diff
@@ -0,0 +1,19 @@
+import { fuzzyMatch } from "../../../utils/strings";
+import { FileElement, ImportTaskpaneState } from "./ImportTaskpane";
+
+
+export const getElementsToDisplay = (importState: ImportTaskpaneState): FileElement[] => {
```
Comment (naterush) [inline mitosheet/src/components/taskpanes/Import/importUtils.tsx:5]: Move utilities here

**Review by naterush — COMMENTED** [review]
Cleanup!

File: mitosheet/mitosheet/step_performers/sort.py [hunk 12]
```diff
@@ -61,7 +62,13 @@ def execute( # type: ignore
 
         try: 
             pandas_start_time = perf_counter()
-            new_df = prev_state.dfs[sheet_index].sort_values(by=column_header, ascending=(sort_direction == ASCENDING), na_position=('first' if sort_direction == ASCENDING else 'last'))
+            if sort_direction != SORT_DIRECTION_NONE:
+                new_df = prev_state.dfs[sheet_index].sort_values(by=column_header, ascending=(sort_direction == SORT_DIRECTION_ASCENDING), na_position=('first' if sort_direction == SORT_DIRECTION_ASCENDING else 'last'))
+            else:
+                # We notably let the user sort by the "none" direction, which effectively allows the user
+                # to unapply the sort by toggling the sort button after they click it once
+                new_df = prev_state.dfs[sheet_index]
```
Comment (aarondr77) [inline mitosheet/mitosheet/step_performers/sort.py:70]: Nice!

**Review by aarondr77 — COMMENTED** [review]
Code looks good. Now to test on windows!

**Pull request merged**